### PR TITLE
Locations 1 - Basic site spawning and management

### PR DESCRIPTION
**Pull request by Spoffy**

### CODE REVIEW

File: game/functions/systems/behaviour_trees/nodes/global/fn_btree_decorator_isInCombat.sqf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
```
Comment (veteran29) [inline game/functions/systems/behaviour_trees/nodes/global/fn_btree_decorator_isInCombat.sqf]: Does not look like this file belongs in this PR.

File: game/functions/systems/missions/client/internal/NOTES.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
+(Done) create
```
Comment (veteran29) [inline game/functions/systems/missions/client/internal/NOTES.txt]: This seems random too.

File: game/functions/systems/sites/server/fn_sites_addSiteType.sqf [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        _siteType - Details for the site, as returned by vgm_s_fnc_sites_getTemplate [HashMap]
+
+    Returns:
+        Nothing
+
+    Example(s):
+        private _mySite = [] call vgm_s_fnc_sites_getTemplate;
+        _mySite set ["id", "vgm_bunker"];
+        // ...etc...
+        [_mySite] call vgm_s_fnc_sites_addSiteType;
```
Comment (veteran29) [inline game/functions/systems/sites/server/fn_sites_addSiteType.sqf]: Example does not match function signature/usage.

File: game/functions/systems/sites/server/fn_sites_preInit.sqf [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    Parameter(s):
+        N/A
+
+    Returns:
+        N/A
+
+    Example(s):
+        N/A
+ */
+
```
Comment (veteran29) [inline game/functions/systems/sites/server/fn_sites_preInit.sqf:20]:
```suggestion

if (!isServer) exitWith {};

```

File: game/functions/systems/sites/server/fn_sites_postInit.sqf [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    Parameter(s):
+        N/A
+
+    Returns:
+        N/A
+
+    Example(s):
+        N/A
+ */
+
```
Comment (veteran29) [inline game/functions/systems/sites/server/fn_sites_postInit.sqf:20]:
```suggestion
if (!isServer) exitWith {};

```

File: game/functions/functions_server.hpp [hunk 6]
```diff
@@ -185,6 +185,36 @@ class vgm_s
...
+        class sites_addSiteType {};
+        class sites_delete {};
+        class sites_getSiteType {};
+        class sites_getTemplate {};
+        class sites_loadSiteTypesFromConfig {};
+        class sites_spawn {};
+        class sites_spawnRandomFortifications {};
+    };
+
+    class site_types
```
Comment (veteran29) [inline game/functions/functions_server.hpp:210]: Not 100% fond of the name, I feel like `site_templates` or `site_configs` would reflect better the purpose of these.

**Review by veteran29 — CHANGES_REQUESTED** [review]

File: game/functions/functions_server.hpp (hunk 6 above)
Comment (Spoffy) [reply game/functions/functions_server.hpp:210]: I'll refactor this another PR, it'd cause annoying conflicts to do it in this one at this point! Happy with site_templates

**Review by veteran29 — APPROVED** [review]
LGTM

**Pull request merged**